grr/server/grr_response_server/gui/ui/components/code_editor/code_editor.ts

```diff
@@ -74,14 +60,155 @@ export class CodeEditor implements AfterViewInit, ControlValueAccessor {
     this.editor = CodeMirror.fromTextArea(this.editorTarget.nativeElement, {
       value: '',
       mode: 'text/x-sqlite',
-      theme: 'idea',
+      theme: 'neo',
       extraKeys: {'Ctrl-Space': 'autocomplete'},
       lineNumbers: true,
       lineWrapping: true,
     });
 
     this.editor.on('change', () => {
       this.announceValueChanged(this.editorValue);
+      this.editorValueChanges$.next(this.editorValue);
     });
   }
 }
+
+/**
+ * Extends the core editor with ControlValueAccessor functoinality.
+ * This allows the code editor to be used as a Angular form field.
+ */
+class EditorWithCva extends CodeEditorCore implements ControlValueAccessor {
+  writeValue(value: string): void {
+    this.editorValue = value;
+  }
+  registerOnChange(fn: OnChangeFn): void {
+    this.announceValueChanged = fn;
+  }
+  registerOnTouched(): void { }
+}
+
+/**
+ * Extends the editor with MatFormFieldControl functionality.
+ * This allows the editor to be used inside <mat-form-field></mat-form-field>
+ * tags.
+ */
+class EditorWithCvaAndMffc extends EditorWithCva
+    implements MatFormFieldControl<string>, OnDestroy {
+  private static uniqueNumber = 0;
+
+  private readonly unsubscribe$ = new Subject<void>();
+
+  readonly controlType = 'code-editor';
+
+  @HostBinding()
+  get id(): string {
+    const uniqueId = `${this.controlType}-${EditorWithCvaAndMffc.uniqueNumber}`;
+    EditorWithCvaAndMffc.uniqueNumber += 1;
+    return uniqueId;
```

Comment:
I'd expect a getter to return the same value always and not update internal state when it is called. If this assumption is false, please document in a comment.

Depending on how IDs are used, you can also use a GUID to avoid keeping `uniqueNumber` as state.